### PR TITLE
Enable coverage report in PRs

### PR DESCRIPTION
**Pull request by westin-m**
Use github actions to report code coverage in PRs.

### CODE REVIEW

**Review by jmprieur — APPROVED** [review]
LGTM
thanks @westin-m

**Comment by jmprieur** [comment]
we can probably remove the test assemblies?
image

**Comment by westin-m** [comment]
> we can probably remove the test assemblies?

I tried a few different things in runsettings, etc. and other places that were all recommended by docs and community posts, but nothing seems to work.

**Review by jennyf19 — APPROVED** [review]

**Review by kllysng — APPROVED** [review]

File: .github/workflows/dotnetcore.yml [hunk 1]
```diff
@@ -30,20 +34,45 @@ jobs:
...
         dotnet-version: 6.0.x
-
+    
     - name: Setup .NET 8.0.x
       uses: actions/setup-dotnet@v4.0.0
       with:
-        dotnet-version: 8.0.x
-
+        dotnet-version: 8.0.x     
+
```
Comment (msbw2) [inline .github/workflows/dotnetcore.yml]: NIT: Remove extra spacing.
```suggestion

```

File: .github/workflows/dotnetcore.yml [hunk 2]
```diff
@@ -30,20 +34,45 @@ jobs:
       uses: actions/setup-dotnet@v4.0.0
       with:
         dotnet-version: 6.0.x
-
+
```
Comment (msbw2) [inline .github/workflows/dotnetcore.yml]:
```suggestion

```

File: .github/workflows/dotnetcore.yml [hunk 3]
```diff
@@ -30,20 +34,45 @@ jobs:
...
       with:
         dotnet-version: 6.0.x
-
+    
     - name: Setup .NET 8.0.x
       uses: actions/setup-dotnet@v4.0.0
       with:
-        dotnet-version: 8.0.x
-
+        dotnet-version: 8.0.x
```
Comment (msbw2) [inline .github/workflows/dotnetcore.yml]:
```suggestion
        dotnet-version: 8.0.x
```

**Pull request merged**